tba-unit-tests/View Elements/Match/MatchViewModelTests.swift

```diff
@@ -0,0 +1,138 @@
+                "faceTheBossRankingPoint": false
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 1)
+        XCTAssertEqual(subject.blueRPCount, 1)
+    }
+
+    func test_second_rp() {
+        let match = insertMatch(eventKey: "2018inwla")
+        match.breakdown = [
+            "red": [
+                "autoQuestRankingPoint": false,
+                "faceTheBossRankingPoint": true
+            ],
+            "blue": [
+                "autoQuestRankingPoint": false,
+                "faceTheBossRankingPoint": true
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 1)
+        XCTAssertEqual(subject.blueRPCount, 1)
+    }
+
+    func test_both_rp() {
+        let match = insertMatch(eventKey: "2018inwla")
+        match.breakdown = [
+            "red": [
+                "autoQuestRankingPoint": true,
+                "faceTheBossRankingPoint": true
+            ],
+            "blue": [
+                "autoQuestRankingPoint": true,
+                "faceTheBossRankingPoint": true
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 2)
+        XCTAssertEqual(subject.blueRPCount, 2)
+    }
+
+    func test_2018_rp() {
+        let match = insertMatch(eventKey: "2018inwla")
+        match.breakdown = [
+            "red": [
+                "autoQuestRankingPoint": true,
+                "faceTheBossRankingPoint": true
+            ],
+            "blue": [
+                "autoQuestRankingPoint": true,
+                "faceTheBossRankingPoint": true
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 2)
+        XCTAssertEqual(subject.blueRPCount, 2)
+    }
+
+    func test_2017_rp() {
+        let match = insertMatch(eventKey: "2017inwla")
+        match.breakdown = [
+            "red": [
+                "kPaRankingPointAchieved": true,
+                "rotorRankingPointAchieved": true
+            ],
+            "blue": [
+                "kPaRankingPointAchieved": true,
+                "rotorRankingPointAchieved": true
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 2)
+        XCTAssertEqual(subject.blueRPCount, 2)
+    }
+
+    func test_2016_rp() {
+        let match = insertMatch(eventKey: "2016inwla")
+        match.breakdown = [
+            "red": [
+                "teleopDefensesBreached": true,
+                "teleopTowerCaptured": true
+            ],
+            "blue": [
+                "teleopDefensesBreached": true,
+                "teleopTowerCaptured": true
+            ]
+        ]
+
+        let subject = MatchViewModel(match: match)
+        XCTAssertEqual(subject.redRPCount, 2)
+        XCTAssertEqual(subject.blueRPCount, 2)
+    }
```

Comment:
space after here please

Reply:
Added in 9f02a264bb7da94be1164f8501639943760201ef